src/main/java/org/folio/circulation/resources/CheckOutByBarcodeResource.java

```diff
@@ -178,6 +198,39 @@ private CompletableFuture<Result<LoanAndRelatedRecords>> saveCheckOutSessionReco
         succeeded(records)));
   }
 
+  private CompletableFuture<Result<LoanAndRelatedRecords>> acquireLock(LoanAndRelatedRecords records, CheckOutLockRepository checkOutLockRepository, AtomicReference<String> checkOutLockId) {
+    log.debug("acquireLock:: Creating checkout lock {} ", records.getCheckoutLockConfiguration());
+    if (records.isCheckoutLockFeatureEnabled()) {
+      CompletableFuture<CheckOutLock> future = new CompletableFuture<>();
+      checkOutLockRepository.createLockWithRetry(0, future, records);
+      return future.handle((res, err) -> {
+        if (err != null) {
+          log.error("acquireLock:: Unable to acquire lock for item {} ", records.getItem().getBarcode(), err);
+          return Result.failed(ValidationErrorFailure.singleValidationError("unable to acquire lock", "", ""));
+        }
+        checkOutLockId.set(res.getId());
+        return Result.succeeded(records);
+      });
+    }
+    return completedFuture(Result.succeeded(records));
+  }
+
+  private CompletableFuture<Result<LoanAndRelatedRecords>> validateItemLimitBasedOnLockFeatureFlag(LoanAndRelatedRecords records, CheckOutValidators validators, CirculationErrorHandler errorHandler) {
```

Comment:
This line (and some tohers) seems to exceed 100 characters. Again, it's not set in stone, but long lines are hard to read. Sometimes we can go a little over 100 but definitely under 120.

Reply:
I have fixed this wherever it is applicable.